pallets/msa/src/lib.rs

```diff
@@ -161,6 +180,12 @@ pub mod pallet {
 		/// calculated value.
 		#[pallet::constant]
 		type MaxSignaturesStored: Get<Option<u32>>;
+
+		/// The origin that is allowed to create providers
+		type CreateProviderOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::AccountId>;
+
+		/// The origin that is allowed to create providers via governance
+		type CreateProviderViaGovernanceOrigin: EnsureOrigin<Self::RuntimeOrigin>;
```

Comment:
I think we should consider adding feature flags here and on the extrinsic so that they do not appear in the Frequency production if not needed.

Reply:
If we are able to conditionally compile the extrinsic based on the `frequency` feature flag, we can remove the `CreateProviderOrigin` type.  However, we'd have to also verify that the removal of the extrinsic doesn't break any other integrations, etc..

Reply:
Hmm... We have a rococo only pallet (the sudo pallet), but I don't think we want to have an extrinsic skipped in mainnet. The `pallet::call_index` makes it at least a possibility to do however.

That said, if we conditionally compiled it, we would need to do separate benchmarks or modify the generated benchmarks which is a bad idea.

Finally, this is a short-term fix (see Long Term: User Journey in the issue: https://github.com/LibertyDSNP/frequency/issues/321)

Reply:
This was not possible but more importantly, we are planning to keep `create_provider() `in the future.